test/Main.purs

```diff
@@ -173,24 +168,20 @@ main = runTest do
       unp (Dictionary dict) = dict
 
     test "All imperial units" do
-      for_ (unp imperialUnitDict) $ \(unit ==> unitStrs) → do
+      for_ (unp imperialUnitDict) \(unit ==> unitStrs) →
         allParseAs (Expression (Unit  unit))
                    unitStrs
 
     test "All normal units" do
-      for_ (unp normalUnitDict) $ \(unit ==> unitStrs) → do
+      for_ (unp normalUnitDict) \(unit ==> unitStrs) →
         allParseAs (Expression (Unit  unit))
                    unitStrs
 
-    for_ (unp prefixDict) $ \(prefix ==> prefixStrs) → do
+    for_ (unp prefixDict) \(prefix ==> prefixStrs) →
       test ("Testing all units with prefix: " <> intercalate ", " prefixStrs) do
-        for_ (unp normalUnitDict) $ \(unit ==> unitStrs) → do
-          let allCombinations = do
-                p <- prefixStrs
-                u <- unitStrs
-                pure (p <> u)
+        for_ (unp normalUnitDict) \(unit ==> unitStrs) →
           allParseAs (Expression (Unit (prefix unit)))
-                     allCombinations
+                     ((<>) <$> prefixStrs <*> unitStrs)
```

Comment:
I'm not so sure about this change; it may be less clear than the original monadic code. What do you think?

Reply:
I think it's good - thank you!